Demo/Demo/Components/Ribbon/RibbonDemoView.swift

```diff
@@ -5,73 +5,42 @@
 import FinniversKit
 
 public class RibbonDemoView: UIView {
-    public lazy var headerLabel: Label = {
-        let label = Label(style: .title1)
-        label.translatesAutoresizingMaskIntoConstraints = false
-        label.text = "Status ribbon"
-        return label
-    }()
+    private lazy var stackView = UIStackView(axis: .vertical, spacing: .spacingS, withAutoLayout: true)
+
+    private let ribbonViews: [UIView] = [
+        RibbonView(style: .success, with: "Success"),
+        RibbonView(style: .error, with: "Error"),
+        RibbonView(style: .warning, with: "Warning"),
+        RibbonView(style: .default, with: "Default"),
+        RibbonView(style: .disabled, with: "Disabled"),
+        RibbonView(style: .sponsored, with: "Sponsored"),
+        IconRibbonView(style: .success, with: UIImage(named: .blinkRocketMini))
+    ]
 
     override init(frame: CGRect) {
         super.init(frame: frame)
-
         setup()
     }
 
     public required init?(coder aDecoder: NSCoder) { fatalError() }
 
     private func setup() {
-        let successRibbonView = RibbonView(style: .success)
-        let errorRibbonView = RibbonView(style: .error)
-        let warningRibbonView = RibbonView(style: .warning)
-        let defaultRibbonView = RibbonView(style: .default, with: "Default")
-        let disabledRibbonView = RibbonView(style: .disabled, with: "Disabled")
-        let sponsoredRibbonView = RibbonView(style: .sponsored, with: "Sponsored")
-
-        let interimSpacing: CGFloat = 16
-        let lineSpacing: CGFloat = 8
-        let margin: CGFloat = 16
-
-        successRibbonView.title = "Success"
-        errorRibbonView.title = "Error"
-        warningRibbonView.title = "Warning"
+        let arrangedStackViews = ribbonViews.map { ribbonView -> UIStackView in
+            ribbonView.translatesAutoresizingMaskIntoConstraints = false
+            ribbonView.setContentHuggingPriority(.required, for: .horizontal)
 
-        defaultRibbonView.translatesAutoresizingMaskIntoConstraints = false
-        successRibbonView.translatesAutoresizingMaskIntoConstraints = false
-        warningRibbonView.translatesAutoresizingMaskIntoConstraints = false
-        errorRibbonView.translatesAutoresizingMaskIntoConstraints = false
-        disabledRibbonView.translatesAutoresizingMaskIntoConstraints = false
-        sponsoredRibbonView.translatesAutoresizingMaskIntoConstraints = false
+            let rowStackView = UIStackView(axis: .horizontal, withAutoLayout: true)
+            rowStackView.addArrangedSubviews([ribbonView, UIView(withAutoLayout: true)])
+            return rowStackView
+        }
```

Comment:
mmm, I was touching ribbon view in another PR. I was thinking on potentially set the content hugging .required as part of the internal setup of ribbon view since I would imagine we wouldn't need it expandable.

then I guess this code may not be needed. It also strikes me like we could use `UILayoutGuide` here, like @hakloev taught us :D

Reply:
We seem to use `setContentHuggingPriority` a lot for ribbons, so that may be a good idea! It would require a lot of changes, so we should probably add a task for it.

Regarding the layout guides: do you have any input on how they could help us here, @hakloev?🙂

Reply:
Not actually, here I would have just done: 

```
private func setup() {
    stackView.alignment = .leading

    stackView.addArrangedSubviews(ribbonViews)
    addSubview(stackView)

    NSLayoutConstraint.activate([
        stackView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .spacingS),
        stackView.topAnchor.constraint(equalTo: topAnchor, constant: .spacingS),
    ])
}
```

Reply:
The default `UIStackView.Alignment` is `.fill`, so the perpendicular axis to the distribution will fail unless it's set to `.leading`

Reply:
Wow, awesome!😻 Thanks!